rosbridge_library/src/rosbridge_library/internal/message_conversion.py

```diff
@@ -306,24 +303,24 @@ def _to_binary_inst(msg):
 def _to_time_inst(msg, rostype, inst=None):
     # Create an instance if we haven't been provided with one
 
-    if rostype == "time" and msg == "now":
+    if rostype == "builtin_interfaces/Time" and msg == "now":
         return ROSClock().now().to_msg()
 
     if inst is None:
-        if rostype == "time":
-            inst = rclpy.time.Time().to_msg()
-        elif rostype == "duration":
-            inst = rclpy.duration.Duration().to_msg()
+        if rostype == "builtin_interfaces/Time":
+            inst = Time().to_msg()
+        elif rostype == "builtin_interfaces/Duration":
+            inst = Duration().to_msg()
         else:
             return None
 
     # Copy across the fields, try ROS1 and ROS2 fieldnames
-    for field in ["secs", "nsecs", "sec", "nanosec"]:
-        try:
-            if field in msg:
-                setattr(inst, field, msg[field])
-        except TypeError:
-            continue
+    for field in ["sec", "secs"]:
+        if field in msg:
+            setattr(inst, "sec", msg[field])
+    for field in ["nanosec", "nsecs"]:
+        if field in msg:
+            setattr(inst, "nanosec", msg[field])
```

Comment:
Since `inst` is ROS 2 type, I believe we must force converting to ROS 2 fields here.